src/main/java/web/cinema/dao/impl/ShoppingCardDaoImpl.java

```diff
@@ -0,0 +1,88 @@
+package web.cinema.dao.impl;
+
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import web.cinema.dao.ShoppingCartDao;
+import web.cinema.exception.DataProcessingException;
+import web.cinema.lib.Dao;
+import web.cinema.model.ShoppingCart;
+import web.cinema.model.User;
+import web.cinema.util.HibernateUtil;
+
+@Dao
+public class ShoppingCardDaoImpl implements ShoppingCartDao {
+    private static final Logger LOGGER = Logger.getLogger(ShoppingCardDaoImpl.class);
+
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            LOGGER.info("shopping cart was succeed added to the db");
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t insert shopping cart entity", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Query query = session.createQuery("from ShoppingCart c "
+                    + "left join fetch c.tickets Ticket "
+                    + "left join fetch c.user User "
+                    + "where c.user = :user", ShoppingCart.class);
```

Comment:
you already have user here. Do you need to fetch it one more time?

Reply:
i see this case on QnA with boroda and do the same